benchmarks/time_solve_models.py

```diff
@@ -36,15 +353,181 @@ def setup(self, solve_first):
         tmax = 4000 / c_rate
         nb_points = 500
         self.t_eval = np.linspace(0, tmax, nb_points)
-        prepare_model(self.model)
+        prepare_model_Ecker2015(self.model)
+        if solve_first:
+            solve_model_once(self.model, TimeSolveSPMEcker2015.solver, self.t_eval)
+
+    def time_solve_model(self, solve_first):
+        TimeSolveSPMEcker2015.solver.solve(self.model, t_eval=self.t_eval)
+
+
+class TimeSolveSPMeMarquis2019:
+    params = [True, False]
+    solver = pybamm.CasadiSolver()
+
+    def setup(self, solve_first):
+        self.model = pybamm.lithium_ion.SPMe()
+        c_rate = 1
+        tmax = 4000 / c_rate
+        nb_points = 500
+        self.t_eval = np.linspace(0, tmax, nb_points)
+        prepare_model_Marquis2019(self.model)
+        if solve_first:
+            solve_model_once(self.model, TimeSolveSPMeMarquis2019.solver, self.t_eval)
+
+    def time_solve_model(self, solve_first):
+        TimeSolveSPMeMarquis2019.solver.solve(self.model, t_eval=self.t_eval)
+
+
+class TimeSolveSPMeORegan2021:
+    params = [True, False]
+    solver = pybamm.CasadiSolver()
+
+    def setup(self, solve_first):
+        self.model = pybamm.lithium_ion.SPMe()
+        c_rate = 1
+        tmax = 4000 / c_rate
+        nb_points = 500
+        self.t_eval = np.linspace(0, tmax, nb_points)
+        prepare_model_ORegan2021(self.model)
+        if solve_first:
+            solve_model_once(self.model, TimeSolveSPMeORegan2021.solver, self.t_eval)
+
+    def time_solve_model(self, solve_first):
+        TimeSolveSPMeORegan2021.solver.solve(self.model, t_eval=self.t_eval)
+
+
+class TimeSolveSPMeNCA_Kim2011:
+    params = [True, False]
+    solver = pybamm.CasadiSolver()
+
+    def setup(self, solve_first):
+        self.model = pybamm.lithium_ion.SPMe()
+        c_rate = 1
+        tmax = 4000 / c_rate
+        nb_points = 500
+        self.t_eval = np.linspace(0, tmax, nb_points)
+        prepare_model_NCA_Kim2011(self.model)
+        if solve_first:
+            solve_model_once(self.model, TimeSolveSPMeNCA_Kim2011.solver, self.t_eval)
+
+    def time_solve_model(self, solve_first):
+        TimeSolveSPMeNCA_Kim2011.solver.solve(self.model, t_eval=self.t_eval)
+
+
+class TimeSolveSPMePrada2013:
+    params = [True, False]
+    solver = pybamm.CasadiSolver()
+
+    def setup(self, solve_first):
+        self.model = pybamm.lithium_ion.SPMe()
+        c_rate = 1
+        tmax = 4000 / c_rate
+        nb_points = 500
+        self.t_eval = np.linspace(0, tmax, nb_points)
+        prepare_model_Prada2013(self.model)
+        if solve_first:
+            solve_model_once(self.model, TimeSolveSPMePrada2013.solver, self.t_eval)
+
+    def time_solve_model(self, solve_first):
+        TimeSolveSPMePrada2013.solver.solve(self.model, t_eval=self.t_eval)
+
+
+class TimeSolveSPMeAi2020:
+    params = [True, False]
+    solver = pybamm.ScikitsDaeSolver()
```

Comment:
Sorry, I didn't explain myself very well on Slack. In the benchmarks we should use the same solver (with same settings, like dt_max) for all tests, so results are comparable. I suggested trying Scikits to check if the model solved fine. If it does, then you can go and fiddle with CasADI. We still don't know why this happens, but sometimes changing the tmax (like to `3800 / c_rate` or `4200 / c_rate`) makes the solver work. If you can't get it to work straight away, just comment the benchmark out and we can skip it for the time being.